cli/src/android/update.ts

```diff
@@ -350,6 +350,65 @@ ext {
     join(config.android.cordovaPluginsDirAbs, 'cordova.variables.gradle'),
     cordovaVariables,
   );
+  await kotlinNeededCheck(config, cordovaPlugins, pluginsGradlePath);
+}
+
+async function kotlinNeededCheck(
+  config: Config,
+  cordovaPlugins: Plugin[],
+  pluginsGradlePath: string,
+) {
+  let kotlinNeeded =
+    config.app.extConfig.cordova?.preferences?.GradlePluginKotlinEnabled ===
+    'true';
+  if (!kotlinNeeded) {
+    for (const plugin of cordovaPlugins) {
+      let platforms: any[] = plugin.xml.platform;
```

Comment:
we have a helper function to get the platform, better use that, see an example of how to get the source-files, it's similar, but for consistency
https://github.com/ionic-team/capacitor/blob/main/cli/src/android/update.ts#L360-L365